contributing/http_server.md

```diff
@@ -0,0 +1,96 @@
+# HTTP Server
```

Comment:
similarly here, we need to explain why someone would need this page's information. Maybe there should be a "Contributing" homepage that ties everything together.